go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -5817,6 +5828,13 @@ var DiffSummaryTableFunctionScriptTests = []queries.ScriptTest{
 				Query:       "SELECT * from dolt_diff_summary(@Commit1, @Commit5, 't');",
 				ExpectedErr: sql.ErrTableNotFound,
 			},
+			{
+				Query: `
+SELECT *
+from dolt_diff_summary(@Commit3, @Commit4, 't') 
+inner join t as of @Commit3 on rows_unmodified = t.pk;`,
```

Comment:
Are there any other join types worth testing?

Reply:
The diff table function isn't going to join against itself, but the fix is a bit involved, not sure this is the right place. I will make an issue. Otherwise I made every table function a generic join node leaf, I am not super worried about the execution logic having issues. Can you think of any table functions (other than json_table) whose join order can't be rearranged?

Reply:
https://github.com/dolthub/dolt/issues/4798

Reply:
Nice! I agree joining to itself doesn't seem high priority. It should be fine to rearrange the join order for any of the (non json_table) table functions we have.